### ISSUE TITLE
Query has unexpected behavior with add()

### ISSUE THREAD

**Issue by yandodov** [getkirby/kirby#2511]
**Describe the bug**  

This query returns the correct set of pages:

```
page.children
```

This one also returns the correct set:

```
site.children.template('modules').index
```

However, if I try to combine them:

```
page.children.add(site.children.template('modules').index)
```

I get the error:

>Your query must return a set of pages

What's also strange is that if I add a `listed` filter to the mix:

```
page.children.add(site.children.template('modules').index).listed
```

...I get the correctly filtered result of `page.children`, i.e. it's the same as if I used

```
page.children.listed
```

**To Reproduce**  
1. Add a bunch of pages with different templates
2. Try the queries above

**Expected behavior**  
The result of the problematic query should be the combined result of the first two queries that work as expected.

**Kirby Version**  
3.3.4

**Comment by afbora** [comment]
This is an expected behavior. The `add()` method [returns the class itself](https://github.com/getkirby/kirby/blob/3.3.4/src/Cms/Pages.php#L47-L67) after adding to the collection, not the children/items. Therefore, after using the `add()` method, you have to use a method again to get the pages like your usage. For example;

````yaml
  page.children.add(site.children.template('modules').index).index
  page.children.add(site.children.template('modules').index).children
  page.children.add(site.children.template('modules').index).listed
  page.children.add(site.children.template('modules').index).unlisted
  page.children.add(site.children.template('modules').index).published
````

**Comment by afbora** [comment]
Feel free to reopen the issue if you think this is a bug.

**Comment by distantnative** [comment]
Actually it returns the altered pages collection wo this shouldn’t be necessary

**Comment by afbora** [comment]
````php
return $this;
````
I'm confused 🤔

**Comment by afbora** [comment]
@distantnative Hmm, you are right, the `Query` class cannot parse correctly when nested round brackets: `.zero(first.two("hello").three)`

**Comment by afbora** [comment]
I think the issue is about opening and closing brackets. Trying to get the first ending round brackets when there are nested round brackets.

Similar kirbytags issue: https://github.com/getkirby/ideas/issues/10

**Comment by distantnative** [comment]
It used to be fine... maybe the issue is that something still follows after the inner brackets

**Comment by distantnative** [comment]
@lukasbestle since you just took on the Query class - you have an idea why thIs happens (or maybe you already fixed it?)

**Comment by lukasbestle** [comment]
This will be fixed in Kirby 3.4.

@hdodov Could you please test the updated code from #2639 with your original query that didn't work before?

**Comment by bastianallgeier** [comment]
✅
